saleor/core/extensions/plugin.py

```diff
@@ -0,0 +1,84 @@
+from typing import TYPE_CHECKING, List, Union
+
+from django_countries.fields import Country
+from prices import Money, MoneyRange, TaxedMoney, TaxedMoneyRange
+
+from ..taxes import TaxType
+
+if TYPE_CHECKING:
+    from ...checkout.models import Checkout, CheckoutLine
+    from ...product.models import Product
+    from ...account.models import Address
+    from ...order.models import OrderLine, Order
+
+
+class BasePlugin:
+    """Abstract class for storing all methods available for any plugin."""
+
+    def calculate_checkout_total(
```

Comment:
We have four "calculate_checkout_" functions that calculate some amounts related to checkout. Do we prefer this granularity over single "calculate_checkout_totals" function?

Reply:
I was thinking about this. Multiple methods will be better than single. We have graphQL API which can query only single value. We calculate only the required value instead of all values.
After I will move vatlayer and avalara to the manager we can discuss it one more time.

Reply:
I was thinking: shouldn't we extract ex. _checkout_ related methods to subclass or other separate thing that residues in relevant place?
Same for other domain parts.

Reply:
Yeap, we were discussing it. We will do that in further steps. For now, we want to introduce a single manager but I am still keeping that in mind ;)